crates/oxc_allocator/src/arena.rs

```diff
@@ -5,19 +5,22 @@ use std::{
     self,
     fmt::{self, Debug, Formatter},
     hash::{Hash, Hasher},
-    ops, ptr,
+    marker::PhantomData,
+    ops::{self, Deref},
+    ptr::{self, NonNull},
 };
 
-use bumpalo::collections;
+use allocator_api2::vec;
 pub use bumpalo::collections::String;
+use bumpalo::Bump;
 use serde::{ser::SerializeSeq, Serialize, Serializer};
 
 use crate::Allocator;
 
 /// A Box without Drop.
 /// This is used for over coming self-referential structs.
 /// It is a memory leak if the boxed value has a `Drop` implementation.
-pub struct Box<'alloc, T: ?Sized>(pub &'alloc mut T);
+pub struct Box<'alloc, T: ?Sized>(NonNull<T>, PhantomData<&'alloc T>);
```

Comment:
It should be `PhantomData<(T, &'alloc ())>` or something like this.
It should convey that `Box` owns `T`

Reply:
Landing in https://github.com/oxc-project/oxc/pull/2952